### PR TITLE
88 refactor all the database code and tests to use the firebase emulator

### PR DESCRIPTION
**Pull request by hmiranda-queiros**
This PR refactors all the code of the database in order to perform the tests with the Firebase emulator instead of the MockDatabase.

In this PR all the unused classes are removed like the "Image" class that is now replaced by the "Post" class.
I implemented new tests to increase the coverage but some classes like the "Follow" class still need to be tested by the persons who wrote it.
In the next PRs it is also important to do more tailored tests in order to reach maximum coverage in all classes. Next goal will be to use the emulator for the Authentication part too.

Due to an unknown problem with the CI, when using join() or get() on futures involving functions that modify the Database content it blocks the tests, so to solve it I use Thread.sleep() instead.

Finally, this PR also provides some fixes on previously failing tests due to OpenAI being non deterministic and also due to sleeps being not long enough.

### CODE REVIEW

File: app/src/main/java/ch/epfl/culturequest/social/Post.java [hunk 1]
```diff
@@ -1,40 +1,37 @@
...
-
-import ch.epfl.culturequest.database.Database;
 
 public final class Post {
     private String postId;
     private String uid;
     private String imageUrl;
     private String artworkName;
-    private Date date;
+    private long time;
```
Comment (TheGuyWithoutH) [inline app/src/main/java/ch/epfl/culturequest/social/Post.java:14]: Are you sure if we need at some point the date that Date won't be easier to handle, with the day, hour... already accessible?

File: app/src/main/java/ch/epfl/culturequest/social/Profile.java [hunk 2]
```diff
@@ -22,14 +18,10 @@ public class Profile extends Observable {
 
     private String uid, name, username, email, phoneNumber;
     private String profilePicture;
-    private List<Post> images;
+    private ArrayList<Post> posts;
```
Comment (TheGuyWithoutH) [inline app/src/main/java/ch/epfl/culturequest/social/Profile.java:21]: Better to use an interface for attributes as the implementation matters less. But more importantly, I think that we can remove the list of posts from the user so that it is not put in the database as is. We shouldn't fetch all the posts when we get a profile, this is a separate collection.

**Review by TheGuyWithoutH — COMMENTED** [review]
Good and deep refactoring! I will take care of what you mentioned for follow system and likes. I only mentionned small questions I had.

File: app/src/main/java/ch/epfl/culturequest/social/Post.java (hunk 1 above)
Comment (hmiranda-queiros) [reply app/src/main/java/ch/epfl/culturequest/social/Post.java:14]: Date format of Java is not compatible with Firebase, it stores it incorrectly, you can check yourself if you want and you will see that the year for example is stored as "123".
So the trick I found is to store a timestamp instead and if you need the date on the app you just need to do "New Date(time)" locally and you will have it ;)

File: app/src/main/java/ch/epfl/culturequest/social/Profile.java (hunk 2 above)
Comment (hmiranda-queiros) [reply app/src/main/java/ch/epfl/culturequest/social/Profile.java:21]: Yes you are right, I did that in order to help the person who read the class attribute now that we always use it as an ArrayList but at the next meeting we need to decide a convention and refactor everything. Because the declaration of the lists in Post and Profile is quite a mess ...

File: app/src/main/java/ch/epfl/culturequest/social/Post.java (hunk 1 above)
Comment (TheGuyWithoutH) [reply app/src/main/java/ch/epfl/culturequest/social/Post.java:14]: Okay then, let's go with it

File: app/src/main/java/ch/epfl/culturequest/social/Profile.java (hunk 2 above)
Comment (TheGuyWithoutH) [reply app/src/main/java/ch/epfl/culturequest/social/Profile.java]: Maybe the get posts will then fetch the database, so that accessing this method is easy for this type of query.

**Review by TheGuyWithoutH — COMMENTED** [review]
And what do you think of the list of posts in profile as I mentionned earlier?

**Review by TheGuyWithoutH — APPROVED** [review]
Good to go!

File: app/src/main/java/ch/epfl/culturequest/social/Profile.java (hunk 2 above)
Comment (hmiranda-queiros) [reply app/src/main/java/ch/epfl/culturequest/social/Profile.java]: Yes I get it, as it is not related to this PR let's do it on yours about the post then.
